node/forest_libp2p/src/service.rs

```diff
@@ -397,6 +259,384 @@ where
+                trace!("agent_version {}", info.agent_version);
+                trace!("listening_ addresses {:?}", info.listen_addrs);
+                trace!("observed_address {}", info.observed_addr);
+                trace!("protocols {:?}", info.protocols);
+            }
+            identify::Event::Sent { .. } => (),
+            identify::Event::Pushed { .. } => (),
+            identify::Event::Error { .. } => (),
+        },
+        ForestBehaviourEvent::ChainExchange(ce_event) => match ce_event {
+            RequestResponseEvent::Message { peer, message } => match message {
+                RequestResponseMessage::Request {
+                    request,
+                    channel,
+                    request_id: _,
+                } => {
+                    debug!("Received chain_exchange request (peer_id: {:?})", peer);
+                    let db = db.clone();
+                    // TODO: Make make_chain_exchange_response async
+                    if let Err(e) = bh_mut.chain_exchange.send_response(
+                        channel,
+                        make_chain_exchange_response(db.as_ref(), &request).await,
+                    ) {
+                        warn!("Failed to send ChainExchangeResponse: {e:?}");
+                    }
+                }
+                RequestResponseMessage::Response {
+                    request_id,
+                    response,
+                } => {
+                    let tx = cx_request_table.remove(&request_id);
+
+                    // Send the sucessful response through channel out.
+                    if let Some(tx) = tx {
+                        if tx.send(Ok(response)).is_err() {
+                            debug!("RPCResponse receive timed out")
+                        }
+                    } else {
+                        debug!("RPCResponse receive failed: channel not found");
+                    };
+                }
+            },
+            RequestResponseEvent::OutboundFailure {
+                peer,
+                request_id,
+                error,
+            } => {
+                debug!(
+                    "ChainExchange outbound error (peer: {:?}) (id: {:?}): {:?}",
+                    peer, request_id, error
+                );
+
+                let tx = cx_request_table.remove(&request_id);
+
+                // Send error through channel out.
+                if let Some(tx) = tx {
+                    if tx.send(Err(error.into())).is_err() {
+                        debug!("RPCResponse receive failed")
+                    }
+                }
+            }
+            RequestResponseEvent::InboundFailure {
+                peer,
+                error,
+                request_id: _,
+            } => {
+                debug!(
+                    "ChainExchange inbound error (peer: {:?}): {:?}",
+                    peer, error
+                );
+            }
+            _ => {}
```

Comment:
Would be good to log this too for debugging purposes in case this events gets hit

Reply:
Not sure if it will flood the log, original logic does not log here